### DISCUSSION TITLE
Recursive createMany

### DISCUSSION THREAD

**Discussion by Gercov** [prisma/prisma#12174] [Q&A] [answered]
I have a structure like this:
```
> Channel 
> > Posts[]
> > > PostsViews[]
```

Can I create a channel, several posts, and several postViews with one request?

**Comment by austincrim** [comment, marked as answer]
Hey @Gercov 👋 ,

This is not possible in a single Prisma operation. [Our docs](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#createmany) have this note about `createMany`:
> You cannot create or connect relations - you cannot nest create, createMany, connect, connectOrCreate inside a top-level createMany

You could create a `Channel` and many `Posts` in a single query, but you could not nest another `createMany` under the `Posts`. 

```ts
// this is possible
let result = await prisma.channel.create({
  data: {
    name: 'my channel',
    posts: {
      createMany: {
        data: [{ title: 'a post' }, { title: 'another post' }]
      }
    }
  }
})

// this is not allowed
let result = await prisma.channel.create({
  data: {
    name: 'my channel',
    posts: {
      createMany: {
        data: [
          { 
            title: 'another post',
            views: {
              createMany: {...} // cannot nest createMany inside createMany
            }
          }
        ]
      }
    }
  }
})
```

You could use a raw SQL query using [`$executeRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#executeraw) or create the `PostViews` in a separate query.

Let me know if I can clarify anything!

**Reply by Gercov** [reply]
@austincrim Thanks

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
